src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/TestProgressState.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.OutputDevice.Console;
+
+internal class TestProgressState
+{
+    public TestProgressState(int passed, int failed, int skipped, string assemblyName, string? targetFramework, string? architecture, StopwatchAbstraction stopwatch, string? detail)
+    {
+        Passed = passed;
+        Failed = failed;
+        Skipped = skipped;
+        AssemblyName = assemblyName ?? throw new ArgumentNullException(nameof(assemblyName));
```

Comment:
`assemblyName` is declared non-nullable. If you are doing validation because this can be called by something/someone "external", we should also validate the ints to be postiive.

Reply:
also code that was generated and later fixed to be non nullable. Fixed.